File: backend/Domain/SaleRecord.cs
namespace Domain;

public class SaleRecord(
    int unitsSold,
    string sku,
    string organizationId,
    string productId,
    int unitsReturned,
    DateTime date
)
{
    // core
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Sku { get; set; } = sku;
    public DateTime Date { get; set; } = date;

    // numbers
    public int UnitsSold { get; set; } = unitsSold;
    public int UnitsReturned { get; set; } = unitsReturned;

    // relationships
    public string OrganizationId { get; set; } = organizationId;
    public Organization Organization { get; set; } = null!;
    public string ProductId { get; set; } = productId;
    public Product Product { get; set; } = null!;
}

File: ml-service/app/schemas/sale_record.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict, Field

class SaleRecord (BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    id: str = Field(alias="Id")
    sku: str = Field(alias="Sku")
    date: datetime = Field(alias="Date")

    units_sold: int = Field(alias="UnitsSold")
    units_returned: int = Field(alias="UnitsReturned")

    organization_id: str = Field(alias="OrganizationId")
    product_id: str = Field(alias="ProductId")
